packages/hash/api/src/graphql/resolvers/knowledge/page/update-page-contents.ts

```diff
@@ -141,6 +141,7 @@ export const updatePersistedPageContents: ResolverFn<
           block: insertedBlocks[insertCount]!,
           position: action.insertBlock.position,
           updateSiblings: false,
+          insertedById: userModel.entityId,
```

Comment:
This is an argument for the `insertBlock` method, which populates various `createdById`/`updatedById` fields in the methods implementation.

I prefer to be descriptive in this way for methods like this as it may not make sense for a `insertBlock` or `moveBlock` method to accept `createdById` or `updatedById`, as it is not obvious what the method is "creating" and "updating" without looking at its implementation. Abstracting this into a `insertedById`/`movedById` argument makes more sense to me as it is less confusing to the method caller what the field is being used for.

Reply:
You reasoning makes perfect sense to me, I'm just not sure I agree with the solution. I think this inhibits intuition within the code and will make it far too easy for us to wire up IDs the wrong way around. It was like us inconsistently using `ownedBy` and `accountId` and such all over the place, where maybe it "made more sense" in that specific context but reduced comprehension overall.

I'm wondering if we actually should be passing in a user to basically all functions like this, maybe calling it an `actor` or something. Again this feels like we're just really not at the right place with our design at the moment and there's a better solution waiting to be found (outside the scope of this PR) related to authentication contexts or headers or whatever.

Reply:
Perhaps @CiaranMn has thoughts. Or maybe someone can tell me it's unimportant and it's not worth my worry

Reply:
> maybe calling it an actor or something

I like this idea, and prevents method writers from having to come up with variable names such as `movedById`, `sharedById`, etc.

Reply:
Yes, sounds good to me.

It'll also make it easier when we want to implement tracking / logging of _every_ action (not just mutating – e.g. who accessed secret info).

Reply:
Have replaced these parameters with `actorId` for the model classes in https://github.com/hashintel/hash/pull/1197/commits/087afd71ecdc7ee807d71967976da6fc40a95f46

Reply:
I think so

Reply:
Addressed in:
- https://github.com/hashintel/hash/pull/1197/commits/f8db98675b106277843c899033148ac366dd12e3
- https://github.com/hashintel/hash/pull/1197/commits/d1c44199424f89936c7904132ab18a8f0ecda7ec
- https://github.com/hashintel/hash/pull/1197/commits/0460c6925e91dc7f2bc604d925d56a4259c17d20

Reply:
@Alfred-Mountfield Are we good to resolve this thread?